src/squeeze_xen.ml

```diff
@@ -377,7 +377,11 @@ module Domain = struct
 
   (** Set a domain's maxmem. Don't throw an exception if the domain has been destroyed *)
   let set_maxmem_noexn cnx domid target_kib =
-    let maxmem_kib = xen_max_offset_kib (get_domain_type cnx domid) +* target_kib in
+    let offset_kib : int64 =
+      try (get_memory_offset cnx domid)
```

Comment:
Minor point: `(...)` are not required, so:
```
let offset_kib : int64 = try get_memory_offset cnx domid with _ -> 0L
```

Does it make sense to log the exception or do we expect this to be a regular case?

Reply:
No point in logging the exception because it's only going to be raised during domain creation.
Code simplification makes sense, I'll update the PR.